### PR TITLE
fix: asking for 'Raw Material quantity' in Process manufacturing

### PR DESCRIPTION
**Pull request by nmimsnikhil**
ref : https://app.asana.com/0/1198503745878277/1198558307793134

**Start**
image

**Finish**
image

### CODE REVIEW

**Comment by hrwX** [comment]
@nmimsnikhil Fix travis tests

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 1]
```diff
@@ -646,15 +646,26 @@ erpnext.work_order = {
 
 	show_prompt_for_qty_input: function(frm, purpose) {
 		let max = this.get_max_transferable_qty(frm, purpose);
+		let label = __('Qty for {0}', [purpose]);
+		if(frm.doc.manufacturing_type === "Process" && purpose === "Manufacture"){
```
Comment (neilLasrado) [inline erpnext/manufacturing/doctype/work_order/work_order.js]: if(frm.doc.manufacturing_type === "Process"){
label = __('Raw Material Qty for {0}', [purpose]);

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 2]
```diff
@@ -646,15 +646,26 @@ erpnext.work_order = {
 
 	show_prompt_for_qty_input: function(frm, purpose) {
 		let max = this.get_max_transferable_qty(frm, purpose);
```
Comment (neilLasrado) [inline erpnext/manufacturing/doctype/work_order/work_order.js:654]: get_max_transferable_qty will return FG Qty, you need to change it to RM Qty for Process Manufacturing

**Review by neilLasrado — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/work_order/work_order.js (hunk 2 above)
Comment (nmimsnikhil) [reply erpnext/manufacturing/doctype/work_order/work_order.js:654]: we are not using 'get_max_transferable_qty' if manufacturing type is process

Comment (neilLasrado) [reply erpnext/manufacturing/doctype/work_order/work_order.js:654]: Why? The user still needs to know what's the max he can transfer right.

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 3]
```diff
@@ -646,15 +646,26 @@ erpnext.work_order = {
...
+		let label = __('Qty for {0}', [purpose]);
+		if(frm.doc.manufacturing_type === "Process" && purpose === "Manufacture"){
+			label = __('Raw Material Consumed for {0}', [purpose]);
+			max = frm.doc.raw_material_qty;
+		}
+		else if(frm.doc.manufacturing_type === "Process"){
+			label = __('Raw Material Qty for {0}', [purpose]);
+			max = frm.doc.raw_material_qty;
```
Comment (neilLasrado) [inline erpnext/manufacturing/doctype/work_order/work_order.js]: this wont work. What if there are multiple transfers or multiple FG Entries?

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 4]
```diff
@@ -669,10 +683,14 @@ erpnext.work_order = {
 	make_se: function(frm, purpose) {
 		this.show_prompt_for_qty_input(frm, purpose)
 			.then(data => {
+				let qty = data.qty;
+				if(frm.doc.manufacturing_type === "Process"){
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/work_order/work_order.js]:
```suggestion
				if (frm.doc.manufacturing_type === "Process") {
```

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 5]
```diff
@@ -646,15 +646,29 @@ erpnext.work_order = {
...
 			frappe.prompt({
 				fieldtype: 'Float',
-				label: __('Qty for {0}', [purpose]),
+				label: label,
 				fieldname: 'qty',
 				description: __('Max: {0}', [max]),
 				default: max
 			}, data => {
-				max += (max * (frm.doc.__onload.overproduction_percentage || 0.0)) / 100;
+				if(frm.doc.manufacturing_type != "Process"){
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/work_order/work_order.js]:
```suggestion
				if (frm.doc.manufacturing_type != "Process") {
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 6]
```diff
@@ -646,15 +646,29 @@ erpnext.work_order = {
 
 	show_prompt_for_qty_input: function(frm, purpose) {
 		let max = this.get_max_transferable_qty(frm, purpose);
+		let label = __('Qty for {0}', [purpose]);
+
+		if(frm.doc.manufacturing_type === "Process" && purpose === "Manufacture"){
+			label = __('Raw Material Consumed for {0}', [purpose]);
+			max = max * (frm.doc.raw_material_qty / frm.doc.qty);
```
Comment (neilLasrado) [inline erpnext/manufacturing/doctype/work_order/work_order.js]: you must do the calculation within the get_max_transferable_qty function

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**